src/Utils/ObjectMixin.php

```diff
@@ -461,13 +502,64 @@ public static function getSuggestion(array $items, $value)
 	}
 
 
-	private static function parseFullDoc($class, $pattern)
+	private static function parseFullDoc(\ReflectionClass $rc, $pattern)
 	{
-		$rc = new \ReflectionClass($class);
 		do {
 			$doc[] = $rc->getDocComment();
 		} while ($rc = $rc->getParentClass());
 		return preg_match_all($pattern, implode($doc), $m) ? $m[1] : [];
 	}
 
+
+	/**
+	 * Checks if the public non-static property exists.
+	 * @return bool|'event'
+	 * @internal
+	 */
+	public static function hasProperty($class, $name)
+	{
+		static $cache;
+		$prop = & $cache[$class][$name];
+		if ($prop === NULL) {
+			$prop = FALSE;
+			try {
+				$rp = new \ReflectionProperty($class, $name);
+				if ($rp->isPublic() && !$rp->isStatic()) {
+					$prop = $name >= 'onA' && $name < 'on_' ? 'event' : TRUE;
+				}
+			} catch (\ReflectionException $e) {
+			}
+		}
+		return $prop;
+	}
+
+
+	/**
+	 * Returns array of public (static, non-static and magic) methods.
+	 * @return array
+	 * @internal
+	 */
+	public static function & getMethods($class)
```

Comment:
This https://github.com/nette/utils/pull/105/commits/d465062927b7c5cc2a71d0131572338caa125fc5 will not be merged, it is only for idea.